static/js/protein/data_processing_utils.ts

```diff
@@ -592,3 +537,324 @@ export function getProteinDescription(data: GraphNodes): string {
+  interactionDCID: string,
+  sourceDCID: string,
+  returnDCID = false
+): string {
+  // note this also works in the case of a self-interaction
+  const interactionID = idFromDCID(interactionDCID);
+  const sourceID = idFromDCID(sourceDCID);
+  const id = interactionID
+    .replace(`${sourceID}_`, "")
+    .replace(`_${sourceID}`, "");
+  if (returnDCID) return dcidFromID(id);
+  return id;
+}
+
+/**
+ * Given id of the form {protein id}_{species id} (e.g. P53_HUMAN), parse into and return ProteinNode
+ */
+export function nodeFromID(
+  proteinSpeciesID: string,
+  depth: number
+): ProteinNode {
+  // assumes {species id} does not contain _ (true as of 06/22/22)
+  const lastIndex = proteinSpeciesID.lastIndexOf("_");
+  return {
+    depth,
+    id: proteinSpeciesID,
+    name: proteinSpeciesID.slice(0, lastIndex),
+    species: proteinSpeciesID.slice(lastIndex + 1),
+  };
+}
+
+/**
+ * Given interaction data as a list of InteractingProteinType, process into and return in the following format: 
+ * 
+      {
+
+        nodeData : [
+          { id: MECOM_HUMAN, name: "MECOM", species: "HUMAN", depth: 0 },
+          { id: CTBP1_HUMAN, name: "CTBP1", species: "HUMAN", depth: 1 },
+          { id: SUPT16H_HUMAN, name: "SUPT16H", species: "HUMAN", depth: 1 },
+        ],
+
+        linkData : [
+          { source: MECOM_HUMAN, target: CTPB1_HUMAN, score: 0.3 },
+          { source: MECOM_HUMAN, target: SUPT16H_HUMAN, score: 0.7 },
+        ],
+
+      }.
+ */
+export function getProteinInteractionGraphData(
+  data: InteractingProteinType[]
+): InteractionGraphDataNested {
+  // checks if the data is empty or not
+  if (_.isEmpty(data)) {
+    return;
+  }
+
+  // P53_HUMAN is central protein in below examples.
+  // take interaction names of the form P53_HUMAN_ASPP2_HUMAN | ASPP2_HUMAN_P53_HUMAN and parse into ASPP2_HUMAN.
+  const centerNodeID = data[0].parent;
+  let neighbors = data.map(({ name: interactionID, value }) => {
+    // value is confidenceScore
+    const neighborID = getInteractionTarget(
+      interactionID,
+      dcidFromID(centerNodeID)
+    );
+    const nodeDatum = nodeFromID(neighborID, 1);
+    nodeDatum["value"] = value;
+    return nodeDatum;
+  });
+
+  // delete duplicates and self-interactions (will add support for self-interactions later on)
+  const seen = new Set();
+  neighbors = neighbors.filter((node) => {
+    const duplicate = seen.has(node.name);
+    seen.add(node.name);
+    return !duplicate && node.id !== centerNodeID;
+  });
+
+  // descending order of interaction confidenceScore
+  neighbors.sort((n1, n2) => n2.value - n1.value);
+  // consider only top 10 interactions to avoid clutter
+  neighbors = neighbors.slice(0, MAX_INTERACTIONS);
+
+  const centerDatum = nodeFromID(centerNodeID, 0);
+
+  const linkData: InteractionLink[] = neighbors.map((node) => {
+    return {
+      score: node.value,
+      source: centerNodeID,
+      target: node.id,
+    };
+  });
+
+  return {
```

Comment:
Only big change is return type is now `InteractionGraphDataNested` instead of `InteractionGraphData`